api/wallet.go

```diff
@@ -532,6 +532,33 @@ func (api *API) walletUnlockHandler(w http.ResponseWriter, req *http.Request, _
 	WriteError(w, Error{"error when calling /wallet/unlock: " + modules.ErrBadEncryptionKey.Error()}, http.StatusBadRequest)
 }
 
+// walletChangeKeyHandler handles API calls to /wallet/changekey.
+func (api *API) walletChangeKeyHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var originalKey crypto.TwofishKey
+	var newKey crypto.TwofishKey
+
+	originalKeys := encryptionKeys(req.FormValue("encryptionpassword"))
+	if len(originalKeys) != 1 {
+		WriteError(w, Error{"expected one encryption key passed to encryptionpassword"}, http.StatusBadRequest)
+		return
+	}
+	originalKey = originalKeys[0]
+
+	newKeys := encryptionKeys(req.FormValue("newpassword"))
+	if len(newKeys) != 1 {
+		WriteError(w, Error{"expected one encryption key to be passed to newpassword"}, http.StatusBadRequest)
+		return
+	}
+	newKey = newKeys[0]
```

Comment:
Technically the api promise here is that you'll check every key in the set of keys. Since this is creating a new key, probably best to return an error if there is more than one key.